packages/costa/src/runtime.ts

```diff
@@ -292,13 +302,9 @@ export class CostaRuntime {
   async fetchByStream(stream: NodeJS.ReadableStream): Promise<PluginPackage> {
     const fileDir = path.join(constants.PIPCOOK_TMPDIR, generateId());
     const filename = path.join(fileDir, 'pkg.tgz');
+    const writeStream = createWriteStream(filename);
     await mkdirp(fileDir);
-    await new Promise((resolve, reject) => {
-      const writeStream = createWriteStream(filename);
-      stream.pipe(writeStream);
-      stream.on('end', resolve);
-      stream.on('error', reject);
-    });
+    await this.saveFile(stream, writeStream);
```

Comment:
Could you please move the writeStream to saveFile, too?

Reply:
If that, the unit test will be unable.

Reply:
Moved this function to `utils.ts`.